test/e2e/suite/command/push.go

```diff
@@ -233,6 +243,28 @@ var _ = Describe("Remote registry users:", func() {
 			}))
 		})
 
+		It("should pack with image spec v1.0 when --config is used, --artifact-type is not used, and --image-spec set to auto", func() {
+			repo := pushTestRepo("my/test/repo")
+			configType := "my/config/type"
+			tempDir := PrepareTempFiles()
+
+			ORAS("push", RegistryRef(ZOTHost, repo, tag), "--config", fmt.Sprintf("%s:%s", foobar.FileConfigName, configType), foobar.FileBarName, "-v").
+				MatchStatus([]match.StateKey{
+					{Digest: foobar.FileConfigStateKey.Digest, Name: configType},
+					foobar.FileBarStateKey,
+				}, true, 2).
+				WithWorkDir(tempDir).Exec()
+			// validate
+			fetched := ORAS("manifest", "fetch", RegistryRef(ZOTHost, repo, tag)).Exec().Out.Contents()
+			var manifest ocispec.Manifest
+			Expect(json.Unmarshal(fetched, &manifest)).ShouldNot(HaveOccurred())
+			Expect(manifest.Config).Should(Equal(ocispec.Descriptor{
+				MediaType: configType,
+				Size:      int64(foobar.FileConfigSize),
+				Digest:    foobar.FileConfigDigest,
+			}))
```

Comment:
Should confirm that `manifest.Artifact` is not set

Reply:
Added this confirmation in Line 266